tests/functional/fetchGit.sh

```diff
@@ -306,3 +306,33 @@ git -C "$empty" config user.name "Foobar"
 git -C "$empty" commit --allow-empty --allow-empty-message --message ""
 
 nix eval --impure --expr "let attrs = builtins.fetchGit $empty; in assert attrs.lastModified != 0; assert attrs.rev != \"0000000000000000000000000000000000000000\"; assert attrs.revCount == 1; true"
+
+# Test a repo with `eol=crlf`.
+repo="$TEST_ROOT/crlf"
+git init "$repo"
+git -C $repo config user.email "foobar@example.com"
+git -C $repo config user.name "Foobar"
+
+echo -n -e 'foo\nbar\nbaz' > "$repo/newlines.txt"
+echo -n -e 'foo\nbar\nbaz' > "$repo/test.txt"
+git -C "$repo" add newlines.txt test.txt
+git -C "$repo" commit -m 'Add files containing LF line endings'
+echo 'test.txt eol=crlf' > "$repo/.gitattributes"
+git -C "$repo" add .gitattributes
+git -C "$repo" commit -m 'Add eol=crlf to gitattributes'
+narhash=$(nix eval --raw --impure --expr "(builtins.fetchGit { url = \"$repo\"; ref = \"master\"; }).narHash")
+[[ "$narhash" = "sha256-BBhuj+vOnwCUnk5az22PwAnF32KE1aulWAVfCQlbW7U=" ]]
+
+narhash=$(nix eval --raw --impure --expr "(builtins.fetchGit { url = \"$repo\"; ref = \"master\"; applyFilters = true; }).narHash")
+[[ "$narhash" = "sha256-k7u7RAaF+OvrbtT3KCCDQA8e9uOdflUo5zSgsosoLzA=" ]]
+
+
+# Ensure that NAR hash doesn't depend on user configuration.
+rm -rf $TEST_HOME/.cache/nix
+export GIT_CONFIG_GLOBAL="$TEST_ROOT/gitconfig"
+git config --global core.autocrlf true
+narhash=$(nix eval --raw --impure --expr "(builtins.fetchGit { url = \"$repo\"; ref = \"master\"; }).narHash")
```

Comment:
This may succeed due to the fetcher cache.
Maybe run this as the very first test case
Then at least the error will be caught by the second run (without autocrlf), where it produces a hash mismatch.
This _should_ be possible to mitigate with `--tarball-ttl 0`, but I haven't checked.